src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,120 @@
+
+@Dao
+public class BookDaoImpl implements BookDao {
+
+    @Override
+    public Book create(Book book) {
+        String sql = "INSERT INTO books (title, price) VALUES (?, ?)";
+
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(sql,
+                Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setInt(2, book.getPrice());
+
+            int affectedRows = statement.executeUpdate();
+
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Creating book failed, no rows affected.");
+            }
+
+            try (ResultSet generatedKeys = statement.getGeneratedKeys()) {
+                if (generatedKeys.next()) {
+                    book.setId(generatedKeys.getObject(1, Long.class));
+                } else {
+                    throw new DataProcessingException("Creating book failed, no ID obtained.");
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create book " + book, e);
+        }
+
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(sql)) {
+            statement.setLong(1, id);
+
+            try (ResultSet resultSet = statement.executeQuery()) {
+                if (resultSet.next()) {
+                    return Optional.of(getBook(resultSet));
+                }
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find by id for id = " + id, e);
+        }
+
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String sql = "SELECT * FROM books";
+
+        try (PreparedStatement statement = ConnectionUtil
+                        .getConnection().prepareStatement(sql);
+                ResultSet resultSet = statement.executeQuery()) {
+
+            while (resultSet.next()) {
+                books.add(getBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't retrieve all books", e);
+        }
+
+        return books;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sql = "UPDATE books SET price = ?, title = ? WHERE id = ?";
+        try (PreparedStatement statement = ConnectionUtil.getConnection().prepareStatement(sql)) {
+            statement.setInt(1, book.getPrice());
+            statement.setString(2, book.getTitle());
+            statement.setLong(3, book.getId());
+            if (statement.executeUpdate() > 0) {
+                return book;
+            } else {
+                throw new DataProcessingException("No book found with id = " + book.getId(),
```

Comment:
Let's use more informative messages for exceptions. Instead of `No book found with id = ` + book.getId(), we can use `No book found with id = ` + book.getId() + ` for book: ` + book